code/modules/projectiles/guns/syringe_gun.dm

```diff
@@ -95,3 +94,285 @@
 	force = 2 //Also very weak because it's smaller
 	suppressed = 1 //Softer fire sound
 	can_unsuppress = 0 //Permanently silenced
+
+// Not quite a syringe gun, but also not completely unlike one either.
+/obj/item/gun/syringe/rapidsyringe
+	name = "rapid syringe gun"
+	desc = "A syndicate rapid syringe gun based on an archived NanoTrasen prototype. Capable of storing and filling syringes from an internal reservoir. It has a large flap on the side that you can dump a box or bag of syringes into, and a port for filling it with liquid."
+	icon_state = "rapidsyringegun"
+	max_syringes = 14  // full two boxes worth
+	/// Index of possible transfer amounts to select
+	var/transfer_amount_selection = 1
+	/// Maximum size of the internal reservoir
+	var/reservoir_volume = 300
+	/// Possible options for alt-clicking the
+	var/list/possible_transfer_amounts = list(5, 10, 15)
+	/// Amount of reagents to transfer out at once if no syringe is loaded
+	var/reservoir_transfer_amount = 15
+	/// Whether or not we've alerted the user that the reservoir is empty.
+	var/alarmed = FALSE
+	container_type = AMOUNT_VISIBLE | TRANSPARENT  // We'll handle it ourselves tyvm
+
+/obj/item/gun/syringe/rapidsyringe/Initialize(mapload)
+	. = ..()
+	create_reagents(reservoir_volume)
+
+/obj/item/gun/syringe/rapidsyringe/Destroy()
+	QDEL_LIST(syringes)
+
+	if(chambered)
+		if(chambered.BB)
+			QDEL_NULL(chambered.BB)
+		qdel(chambered)
+
+	return ..()
+
+/obj/item/gun/syringe/rapidsyringe/build_reagent_description(mob/user)
+	. = list("<span class='notice'>There's a little window for the internal reservoir.</span>")
+	. += ..()
```

Comment:
Swap these?

Reply:
I wanted it to show like "oh, there's a window. It lets you see that there are x units of y."